CodingStandard.md

```diff
@@ -0,0 +1,1425 @@
+    Widget widget_;
+}
+
+/*
+ *  (Routine Header)
+ */
+inline int Foo::double_bar() const
+{
+    return 2 * bar_;
+}
+
+} // namespace foos
+
+#endif // FOOS_FOO_H
+```
+
+Sections should be ordered public, protected, then private.
+
+Class members should be declared in this order:
+ - Using-declarations, typedefs and enums
+ - Constants (static const/constexpr data members)
+ - Constructors and assignment operators
+ - Destructor
+ - Methods, including static methods
+ - Member variables (except constant data members)
+
+Function parameters should be declared in this order:
+  - Input
+  - Input/Output
+  - Output
+
+
+**Implementation File Layout**
+
+A typical .cpp file should look like this:
+
+```c++
+/*
+ *  (Copyright and File Header)
+ */
+#include "foo.h"
+#include "library.h"
+#include "other.h"
+#include "core.h"
+
+namespace foos {
+
+/*
+ *  (Routine Header)
+ */
+Foo::Foo() : bar_{}, foo_{}, widget_{}
+{
+}
+
+/*
+ *  (Routine Header)
+ */
+Foo::Foo(int bar): bar_{bar}, id_{}, widget_{}
+{
+}
+
+/*
+ *  (Routine Header)
+ */
+Foo::Foo(int bar, int id): bar_{bar}, id_{id}, widget_{}
+{
+}
+
+/*
+ *  (Routine Header)
+ */
+Foo::~Foo() = default;
+
+/*
+ *  (Routine Header)
+ */
+void Foo::compute_bar(int x)
+{
+    bar_ = 5 * x;
+}
+
+} // namespace foos
+
+```
+
+Member function definitions can appear in any order in the .cpp file, but
+should be ordered logically to help the reader.
+
+The #include for the class header should be the first substantive line in the
+source file, following the file header and the copyright block.
+
+Source code should be wrapped at 90 characters.
+
+
+**Inlining**
+
+Trivial member functions can be defined directly inside the class definition
+and do not require header comments.  They should not be marked `inline` as they
+are considered inline by default.
+
+Other simple member functions can be defined in a header file, but they should
+be defined outside the class definition, with header comments.  They should be
+marked `inline`.  This does not guarantee that the code will be inlined by the
+compiler, but rather it prevents violations of the "One Definition Rule" and
+link-time errors.
+
+Template functions defined in a header file should not be marked `inline`
+as they are considered inline by default.
+
+**Special Member Functions**
+
+The special member functions of a class (constructors, copy/move constructors,
+destructors, and copy/move assignment operators) should be declared in the
+class definition and defined below the class definition in the header file if
+they are inline, or in the source file otherwise.
+
+Undesired special member functions should be marked `= delete;` inside the
+public section of the class definition (rather than making them `private`).
+
+Except for simple struct types, default versions of all special member functions
+should be explicitly declared and defined.  Declare them in the class definition
+and define them using `= default;` in the source file.
```

Comment:
It doesn't change anything in the binary file; it just shows that you left out the destructor on purpose because you wanted to use the default behavior.